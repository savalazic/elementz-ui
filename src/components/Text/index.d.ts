export { default, Props } from './Text';
