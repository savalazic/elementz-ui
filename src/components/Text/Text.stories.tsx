import React from 'react';

import { storiesOf } from '@storybook/react';

import Text from './Text';

storiesOf('Text', module).add('with text', () => (
  <Text text="Test storybook" />
));
