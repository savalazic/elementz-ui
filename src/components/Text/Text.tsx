import React from 'react';
import styled from 'styled-components';

const Test = styled.p`
  font-family: sans-serif;
  color: #333;
`;

export const pureFunction = (state: string) => state;
pureFunction('test');

const Text: React.SFC<{ text: string }> = ({ text }) => {
  return (
    <div>
      <Test>{text}</Test>
    </div>
  );
};

export default Text;
