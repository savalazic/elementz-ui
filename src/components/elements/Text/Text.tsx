import React from 'react';
import styled from 'styled-components';

import { truncateStyles } from '../../../utils/style-helpers';

export enum TextTagEnum {
  p = 'p',
  span = 'span',
  strong = 'strong',
  em = 'em',
}

export interface TextProps {
  /** @default p */
  /** tag of text */
  tag?: 'p' | 'span' | 'strong' | 'em';
  /** content */
  children: React.ReactNode;
  /** if text should be uppercased */
  uppercase?: boolean;
  /** if text should be underlined */
  underline?: boolean;
  /** if text should be truncated... */
  truncate?: string;
}

const StyledText = styled.p<TextProps>`
  text-decoration: ${props => props.underline && 'underline'};
  text-transform: ${props => props.uppercase && 'uppercase'};
  ${props => props.truncate && truncateStyles(props.truncate)};
`;

export const Text = ({
  tag = TextTagEnum.p,
  children,
  uppercase,
  underline,
  truncate,
  ...rest
}: TextProps &
  React.HTMLAttributes<HTMLParagraphElement | HTMLSpanElement>) => {
  return (
    <StyledText
      as={tag}
      uppercase={uppercase}
      underline={underline}
      truncate={truncate}
      {...rest}
    >
      {children}
    </StyledText>
  );
};

export default Text;
