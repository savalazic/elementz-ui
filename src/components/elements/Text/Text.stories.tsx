import React from 'react';

import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs/react';

import { infoConfig } from '../../../utils/info-config';

import { Text, TextTagEnum } from './Text';

const stories = storiesOf('Elements|Text', module);
stories.addParameters(infoConfig);

stories
  .add('default', () => {
    const txt = text(
      'Content',
      'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    );
    return <Text>{txt}</Text>;
  })
  .add('span', () => {
    const txt = text(
      'Content',
      'Lorem ipsum dolor sit amet consectetur adipisicing elit. Recusandae,dicta!',
    );
    return <Text tag={TextTagEnum.span}>{txt}</Text>;
  })
  .add('bold', () => {
    const txt = text(
      'Content',
      'Lorem ipsum dolor sit amet consectetur adipisicing elit. Recusandae,dicta!',
    );
    return <Text tag={TextTagEnum.strong}>{txt}</Text>;
  })
  .add('italic', () => {
    const txt = text(
      'Content',
      'Lorem ipsum dolor sit amet consectetur adipisicing elit. Recusandae,dicta!',
    );
    return <Text tag={TextTagEnum.em}>{txt}</Text>;
  })
  .add('uppercase', () => {
    const txt = text(
      'Content',
      'Lorem ipsum dolor sit amet consectetur adipisicing elit. Recusandae,dicta!',
    );
    return <Text uppercase>{txt}</Text>;
  })
  .add('underline', () => {
    const txt = text(
      'Content',
      'Lorem ipsum dolor sit amet consectetur adipisicing elit. Recusandae,dicta!',
    );
    return <Text underline>{txt}</Text>;
  })
  .add('truncate', () => {
    const txt = text(
      'Content',
      'Lorem ipsum dolor sit amet consectetur adipisicing elit. Recusandae,dicta!',
    );
    return <Text truncate="250px">{txt}</Text>;
  })
  .add('truncate with wrapping element', () => {
    const txt = text(
      'Content',
      'Lorem ipsum dolor sit amet consectetur adipisicing elit. Recusandae,dicta!',
    );
    return (
      <div style={{ width: '150px' }}>
        <Text truncate="100%">{txt}</Text>
      </div>
    );
  });
