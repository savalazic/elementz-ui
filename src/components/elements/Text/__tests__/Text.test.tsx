import 'jest-dom/extend-expect';
import React from 'react';
import { render } from 'react-testing-library';

import { Text } from '../Text';

describe('Text component', () => {
  const txt = 'This is mocked text';

  it('should match default snapshot', () => {
    const { container } = render(<Text>{txt}</Text>);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with underlined text', () => {
    const { container } = render(<Text underline>{txt}</Text>);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with uppercased text', () => {
    const { container } = render(<Text uppercase>{txt}</Text>);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with truncated text', () => {
    const { container } = render(<Text truncate="100px">{txt}</Text>);
    expect(container).toMatchSnapshot();
  });

  it('should render paragraph without props', () => {
    const { container } = render(<Text>{txt}</Text>);
    expect(container).toHaveTextContent(txt);
  });
});
