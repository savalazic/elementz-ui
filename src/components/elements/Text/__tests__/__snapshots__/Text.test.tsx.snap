// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component should match default snapshot 1`] = `
<div>
  <p>
    This is mocked text
  </p>
</div>
`;

exports[`Text component should match snapshot with truncated text 1`] = `
<div>
  <p
    style="display: block; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
  >
    This is mocked text
  </p>
</div>
`;

exports[`Text component should match snapshot with underlined text 1`] = `
<div>
  <p
    style="text-decoration: underline;"
  >
    This is mocked text
  </p>
</div>
`;

exports[`Text component should match snapshot with uppercased text 1`] = `
<div>
  <p
    style="text-transform: uppercase;"
  >
    This is mocked text
  </p>
</div>
`;
